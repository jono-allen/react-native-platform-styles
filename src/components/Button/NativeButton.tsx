import { Platform } from 'react-native';

const Button = Platform.select({
  ios: () => require('react-native-ios-kit/src/components/Button').default,
  default: () => require('react-native-paper/src/components/Button').default,
})();

export default Button;
