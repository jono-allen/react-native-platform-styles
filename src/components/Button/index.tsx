import React from 'react';
import { Platform } from 'react-native';

import { Button as PaperButton } from 'react-native-paper';
import { Button as IOSButton } from 'react-native-ios-kit';

type PropTypes = React.ComponentProps<typeof PaperButton | typeof IOSButton>;
type PropsWithOutTheme = Omit<PropTypes, 'theme'>;

const Button = (props: PropsWithOutTheme) => {
  if (Platform.OS === 'ios') {
    return React.createElement(IOSButton, props);
  }
  return React.createElement(PaperButton, props);
};
export default Button;
