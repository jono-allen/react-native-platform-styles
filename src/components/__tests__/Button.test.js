import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';
import { Platform } from 'react-native';

describe('Button', () => {
  afterEach(() => jest.resetModules());

  describe('on default platform', () => {
    beforeAll(() => {
      jest.mock('react-native', () => {
        const RN = jest.requireActual('react-native');

        RN.Platform = () => ({
          select: (objs) => objs.default,
        });

        return RN;
      });
    });

    it('it is android', () => {
      Platform.OS = 'default';
      expect(Platform.OS).toEqual('default');
    });

    it('it renders correctly', () => {
      Platform.OS = 'default';

      console.log(Platform.OS);

      const tree = renderer.create(<Button>Default Button</Button>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
  describe('on ios platform', () => {
    beforeAll(() => {
      jest.mock('react-native', () => {
        const RN = jest.requireActual('react-native');

        RN.Platform = () => ({
          select: (objs) => objs.ios,
          OS: 'default',
        });

        return RN;
      });
    });

    it('it is ios', () => {
      expect(Platform.OS).toEqual('ios');
    });

    it('it renders correctly', () => {
      Platform.OS = 'ios';
      const tree = renderer.create(<Button>IOS Button</Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
