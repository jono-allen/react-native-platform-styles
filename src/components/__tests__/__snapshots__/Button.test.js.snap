// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button on default platform it renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        Default Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button on ios platform it renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#007aff",
          "fontSize": 17,
          "fontWeight": "normal",
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    IOS Button
  </Text>
</View>
`;
